src/components/common/YourCompany/AssociationType.vue

```diff
@@ -0,0 +1,256 @@
+                </template>
+                <v-list>
+                  <v-list-item
+                    id="btn-more-actions-edit"
+                    class="v-list-item"
+                    @click="isEditingAssociationType = true; dropdown = false"
+                  >
+                    <v-list-item-subtitle>
+                      <v-icon small color="primary">mdi-pencil</v-icon>
+                      <span class="drop-down-action ml-1">Change</span>
+                    </v-list-item-subtitle>
+                  </v-list-item>
+                </v-list>
+              </v-menu>
+            </span>
+        </div>
+      </v-col>
+
+      <!-- Editing Mode -->
+      <v-col cols="9" v-if="isEditingAssociationType">
+        <v-select
+          filled
+          id="association-type-input"
+          :items="associationTypeOptions"
+          v-model="selectedAssociationType"
+          :rules="AssociationTypeRules"
+        >
+        </v-select>
+
+        <!-- Done Actions -->
+        <div class="action-btns">
+          <v-btn
+            id="done-btn"
+            large color="primary"
+            @click="submitAssociationTypeChange()"
+          >
+            <span>Done</span>
+          </v-btn>
+          <v-btn
+            id="cancel-btn"
+            large outlined color="primary"
+            @click="isEditingAssociationType = false"
+          >
+            <span>Cancel</span>
+          </v-btn>
+        </div>
+      </v-col>
+    </v-row>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Mixins, Prop, Watch } from 'vue-property-decorator'
+import { Action, Getter } from 'vuex-class'
+import { AssociationTypes } from '@/enums'
+import { CommonMixin, EnumMixin } from '@/mixins'
+import { ActionBindingIF, BusinessInformationIF, EntitySnapshotIF } from '@/interfaces'
+
+@Component({})
+export default class AssociationType extends Mixins(CommonMixin, EnumMixin) {
+  @Prop({ default: false })
+  readonly invalidSection: boolean
+  /** Global getters */
+  @Getter getAssociationType!: AssociationTypes
+  @Getter getBusinessInformation!: BusinessInformationIF
+  @Getter getEntitySnapshot!: EntitySnapshotIF
+  @Getter hasAssociationTypeChanged!: boolean
+
+  /** Global actions */
+  @Action setBusinessInformation!: ActionBindingIF
+
+  /** Select options */
+  readonly associationTypeOptions: Array<any> = [
+    {
+      value: AssociationTypes.COMMUNITY_SERVICE_COOPERATIVE,
+      text: this.associationTypeToDescription(AssociationTypes.COMMUNITY_SERVICE_COOPERATIVE)
+    },
+    {
+      value: AssociationTypes.ORDINARY_COOPERATIVE,
+      text: this.associationTypeToDescription(AssociationTypes.ORDINARY_COOPERATIVE)
+    },
+    {
+      value: AssociationTypes.HOUSING_COOPERATIVE,
+      text: this.associationTypeToDescription(AssociationTypes.HOUSING_COOPERATIVE)
+    }
+  ]
+
+  protected selectedAssociationType = null as AssociationTypes
+  protected isEditingAssociationType = false
```

Comment:
I think you need to do something extra with this property, so that this component is deemed invalid while it's in editing mode.

Reply:
More than 

```
  /** The association type section validity state (when prompted by app). */
  get invalidAssociationTypeSection (): boolean {
    return (this.getComponentValidate && this.isEditingAssociationType)
  }
```

Reply:
That's exactly right, but I don't see code to pass `isEditingAssociationType` from this component to the parent component (YourCompany) where this validity check is done. You need to emit the change.

Reply:
I'll fix this in the next PR